packages/fuel-indexer-database/src/lib.rs

```diff
@@ -75,15 +78,20 @@ impl IndexerConnectionPool {
         match url.scheme() {
             "postgres" => {
                 let pool = attempt_database_connection(|| {
-                    sqlx::Pool::<sqlx::Postgres>::connect(database_url)
+                    let connection_options = PgConnectOptions::from_str(database_url)
+                        .expect("Failed to parse Postgres connection options");
```

Comment:
- Looks like my original comment didn't go through, but mentioned that we wanna start `expect`'ing less and start `return`'ing `Error`s more where we can
- This is not just for this PR but in general -- if you see `expect`s try converting them unless there's a good reason they're there